### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by Unleashedicon**
- Added Redux Toolkit (npm install react-redux @reduxjs/toolkit).

- Created a directory that will contain all Redux logic (/src/redux)

- Configured a Redux store (/src/redux/store.js)

- Defined a slice of state for books that:

> Includes an array of books (initial state: empty array)

> Includes a reducer that adds a book

> Includes a reducer that removes a book

- Defined a slice of state for categories that:

> Includes an array of categories (initial state: empty array)

> Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
```
Comment (Shedrack-Sunday) [inline src/redux/books/booksSlice.js:22]: - [ ] Based on instructions, you're to Include an array of books (initial state) which should be empty array not populated. Kindly follow this :

### Instructions:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Unleashedicon        👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the react format✔️ 
- Beautiful use of redux 👏    
 
  
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by binhussen — APPROVED** [review]
Hi @Unleashedicon , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Added reducers and actions ✔️
- Well starctured code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by prolajumokeoni** [comment]
Hi @Unleashedicon 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong pull request for this pull milestone: Bookstore: use Redux in React components. 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**